icelake/src/types/arrow/from_arrow/to_array.rs

```diff
@@ -0,0 +1,373 @@
+use super::to_primitive::ToPrimitiveValue;
+use crate::types::PrimitiveValue;
+use crate::types::{Any, AnyValue, StructValueBuilder};
+use crate::Result;
+use arrow::array::{
+    Array, Date32Array, Date64Array, Float32Array, Float64Array, Int16Array, Int32Array,
+    Int64Array, Int8Array, UInt16Array, UInt32Array, UInt64Array, UInt8Array,
+};
+use arrow::{
+    array::{ArrayRef, BooleanArray, PrimitiveArray, StructArray},
+    datatypes::ArrowPrimitiveType,
+};
+use std::iter::Iterator;
+pub trait ToArray {
+    fn to_anyvalue_array(&self) -> Result<Vec<Option<AnyValue>>>;
+
+    /// This interface is used to convert the struct array. We pass he target type instead of
+    /// converting it internally is to save the extra cost.
+    fn to_anyvalue_array_with_type(&self, target_type: Any) -> Result<Vec<Option<AnyValue>>>;
```

Comment:
Maybe we don't need this trait? We can implement `TryFrom<PrimitiveArray>` and `TryFrom<BoolArray>` for `Vec<Option<AnyValue>>`, and have two methods to convert `ArrayRef` and `StructArray` for type safety.

Reply:
We can't directly implement TryFrom<PrimitiveArray> becase it don't define in our crate, we will get the error like: 
```
only traits defined in the current crate can be implemented for types defined outside of the crate
   define and implement a trait or new type instead 
```